File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/multiline/comment/direct/Java/C++/s012761786_commented.cpp
#include <iostream>
#include <cmath>
#include <cstdint>

using namespace std;

int main() {
    int64_t n, a, b;
    cin >> n >> a >> b;

    int64_t sa = std::llabs(a - b);
    int64_t ans = 0;

    if (sa % 2 == 0) {
        ans = sa / 2;
    } else {
        int64_t AAA = min((n - a), (a - 1));
        int64_t BBB = min((n - b), (b - 1));

        ans = min(AAA, BBB);

        sa -= 1;
        ans += 1;

        ans += sa / 2;
    }

    cout << ans << endl;

    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/multiline/comment/direct/Java/Go/s012761786_commented.go
package main

import (
	"fmt"
	"math"
)

func main() {
	var n, a, b int64
	fmt.Scan(&n, &a, &b)

	sa := int64(math.Abs(float64(a - b)))

	var ans int64

	if sa%2 == 0 {
		ans = sa / 2
	} else {
		var AAA, BBB int64
		AAA = n - a
		if (a-1) < AAA {
			AAA = a - 1
		}

		BBB = n - b
		if (b-1) < BBB {
			BBB = b - 1
		}

		if AAA < BBB {
			ans = AAA
		} else {
			ans = BBB
		}

		sa -= 1
		ans += 1
		ans += sa / 2
	}

	fmt.Println(ans)
}

// End of Code
